### PR TITLE
perlPackages.LaTeXML: 0.8.5 -> 0.8.6

### PR DESCRIPTION
**Pull request by xworld21**
###### Motivation for this change
Yearly update of LaTeXML: [changes](https://github.com/brucemiller/LaTeXML/releases/tag/v0.8.6).

Among other things, version 0.8.6 fixes some regressions with TeX Live 2021.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by veprbl** [comment]
@GrahamcOfBorg build perl532Packages.LaTeXML perl534Packages.LaTeXML

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 140173` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>perl532Packages.LaTeXML</li>
    <li>perl534Packages.LaTeXML</li>
  </ul>
</details>

File: pkgs/top-level/perl-packages.nix [hunk 1]
```diff
@@ -11353,16 +11353,13 @@ let
...
+    version = "0.8.6";
     src = fetchurl {
       url = "mirror://cpan/authors/id/B/BR/BRMILLER/${pname}-${version}.tar.gz";
-      sha256 = "0dr69rgl4si9i9ww1r4dc7apgb7y6f7ih808w4g0924cvz823s0x";
+      sha256 = "1ccvdq7asxq6iw8x8ihwf5xs2mp7fkwm467xy7g8spkznr8wcacm";
     };
     outputs = [ "out" "tex" ];
-    propagatedBuildInputs = [ ArchiveZip DBFile FileWhich IOString ImageSize JSONXS LWP ParseRecDescent ImageMagick PodParser TextUnidecode XMLLibXSLT ];
-    preCheck = ''
-      rm t/931_epub.t # epub test fails
```
Comment (veprbl) [inline pkgs/top-level/perl-packages.nix:11364]: Still fails

Comment (xworld21) [reply pkgs/top-level/perl-packages.nix:11364]: Oh. Apparently the error is
```
Error in tempdir() using /build/XXXXXXXXXX: Parent directory (/build) is not safe (Directory owned neither by root nor the current user) at /build/LaTeXML-0.8.6/blib/script/../lib/LaTeXML.pm line 196.
```
I'll remove the test as I don't see an obvious way to handle this.

File: pkgs/top-level/perl-packages.nix [hunk 2]
```diff
@@ -11353,15 +11353,15 @@ let
...
+    propagatedBuildInputs = [ ArchiveZip DBFile FileWhich IOString ImageMagick ImageSize JSONXS LWP ParseRecDescent PodParser TextUnidecode XMLLibXSLT ];
+    preCheck = /* epub test fails: File::Temp considers /build insecure */ ''
```
Comment (SuperSandro2000) [inline pkgs/top-level/perl-packages.nix:11363]:
```suggestion
    preCheck = ''
      # epub test fails: File::Temp considers /build insecure
```

File: pkgs/top-level/perl-packages.nix (hunk 1 above)
Comment (veprbl) [reply pkgs/top-level/perl-packages.nix:11364]: This works
```diff
diff --git a/pkgs/top-level/perl-packages.nix b/pkgs/top-level/perl-packages.nix
index 6729ed28bc8..6b74bf81c0c 100644
--- a/pkgs/top-level/perl-packages.nix
+++ b/pkgs/top-level/perl-packages.nix
@@ -11360,8 +11360,11 @@ let
     };
     outputs = [ "out" "tex" ];
     propagatedBuildInputs = [ ArchiveZip DBFile FileWhich IOString ImageMagick ImageSize JSONXS LWP ParseRecDescent PodParser TextUnidecode XMLLibXSLT ];
-    preCheck = /* epub test fails: File::Temp considers /build insecure */ ''
-      rm t/931_epub.t
+    # https://github.com/brucemiller/LaTeXML/issues/1669
+    preCheck = ''
+      for file in ./lib/LaTeXML.pm ./lib/LaTeXML/Post/LaTeXImages.pm ./blib/lib/LaTeXML/Post/LaTeXImages.pm ./blib/lib/LaTeXML.pm; do
+        substituteInPlace "$file" --replace "File::Temp::HIGH" "File::Temp::MEDIUM"
+      done
     '';
     nativeBuildInputs = [ pkgs.makeWrapper ] ++ lib.optional stdenv.isDarwin shortenPerlShebang;
     makeMakerFlags = "TEXMF=\${tex} NOMKTEXLSR";
```

Comment (xworld21) [reply pkgs/top-level/perl-packages.nix:11364]: Thanks for the investigation @veprbl! I suggest we keep the current `rm t/931_epub.t` (with link to the new upstream issue) and backport the `File::Temp::MEDIUM` patch once upstream makes a move. Deviating from upstream on a security setting does not feel right.

Comment (veprbl) [reply pkgs/top-level/perl-packages.nix:11364]: I don't think there is any security going on, see the 22y old bug in File::Temp. But I don't mind keeping the rm.

**Pull request merged**